src/window.rs

```diff
@@ -710,11 +706,29 @@ pub trait Window: AsAny + Send + Sync {
     ///
     /// ## Platform-specific
     ///
-    /// - **iOS:** Returns the [`PhysicalSize`] of the window in screen space coordinates.
     /// - **Web:** Returns the size of the canvas element. _Note: this returns the same value as
     ///   [`Window::surface_size`]._
     fn outer_size(&self) -> PhysicalSize<u32>;
 
+    /// The area of the surface that is unobstructed.
+    ///
+    /// On some devices, especially mobile devices, the screen is not a perfect rectangle, and may
+    /// have rounded corners, notches, bezels, and so on. When drawing your content, you usually
+    /// want to draw your background and other such unimportant content on the entire surface, while
+    /// you will want to restrict important content such as text, interactable or visual indicators
+    /// to the part of the screen that is actually visible; for this, you use the safe area.
+    ///
+    /// The safe area is a rectangle that is defined relative to the origin at the top-left corner
+    /// of the surface, and the size extending downwards to the right. The area will not extend
+    /// beyond [the bounds of the surface][Window::surface_size].
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **Wayland / Android / Orbital:** Unimplemented, returns `((0, 0), surface_size)`.
+    /// - **macOS:** This must be used when using `set_simple_fullscreen` to prevent overlapping the
+    ///   notch on newer devices.
```

Comment:
Shouldn't it generally be used when you move your window behind the notch, etc? Generally speaking, both fullscreen/simple fullscreen and window minimization could result in notch obscuring things.

Should we add a `SafeArea` event telling that it got updates or how it generally supposed to work on macOS? I'd assume that you should call it once you enter fullscreen/maximize/simple fullscreen states?

Reply:
Normal full screen modes place the content below the notch, so it's not an issue there.

> Should we add a `SafeArea` event

I think that'd be overcomplicating it, the user should _always_ handle `safe_area` in their rendering code, most of the time it'll just be a no-op. I included this comment for completeness, but perhaps it's more appropriate to only state it on `set_simple_fullscreen`.

Reply:
>I think that'd be overcomplicating it, the user should always handle safe_area in their rendering code, most of the time it'll just be a no-op.

The point was that you may change the state without changing the size, etc, though, it's a separate issue. Maybe it should just schedule redraw on macOS when you change the state of fullscreen, which is likely the case already.

Reply:
I think we should just schedule a redraw in the odd case where the safe area changes without the surface changing